pathd/path_pcep_controller.c

```diff
@@ -673,20 +752,54 @@ int pcep_thread_event_update_pcc_options(struct ctrl_state *ctrl_state,
 	ctrl_state->pcc_opts = opts;
 	return 0;
 }
-
+int pcep_thread_event_update_best_pce(struct ctrl_state *ctrl_state, int best)
+{
+	PCEP_DEBUG(" recalculating pce precedence ");
+	if (best) {
+		struct pcc_state *best_pcc_state =
+			get_pcc_state(ctrl_state, best);
+		if (best_pcc_state->previous_best != best_pcc_state->is_best) {
```

Comment:
Why not have the id of the best pcc and the id of the previous best in the controller state ?